File: src/models/skill.ts
/** Types cho thư viện Skill — khớp với DTO ở `src-tauri/src/models/skill.rs`. */

/**
 * Danh mục skill — giá trị là `name` của một mục `master_data` có
 * `keygroup = "SKILL_TYPE"` (xem `useSkill.ts` / trang Master Data).
 */
export type SkillType = string;

// Backward compat alias (dùng trong các component cũ)
export type SkillCategory = SkillType;

export interface Skill {
  id: number;
  name: string;
  description: string;
  icon: string;
  category: SkillType;
  stack: string;
  instructions: string;
  tags: string[];
  created_at: string;
  updated_at: string;
}

export const DEFAULT_SKILL_ICON = "pi pi-book";

export interface SkillRequest {
  name: string;
  description: string;
  icon: string;
  category: SkillType;
  stack: string;
  instructions: string;
  tags: string[];
}

/** Gợi ý stack theo danh mục (bổ trợ UI — không phải nguồn dữ liệu category). */
export const STACK_SUGGESTIONS: Record<string, string[]> = {
  general:     [],
  frontend:    ["Vue 3", "React", "Angular", "Svelte", "Next.js", "Nuxt.js", "TypeScript", "Tailwind CSS"],
  backend:     ["Python", "FastAPI", "Django", "Node.js", "NestJS", "Express", "Go", "Rust", "Java", "Spring Boot", "PHP", "Laravel"],
  mobile:      ["React Native", "Flutter", "Swift", "Kotlin", "Expo"],
  devops:      ["Docker", "Kubernetes", "GitHub Actions", "Terraform", "AWS", "GCP", "Azure"],
  translation: ["EN → VI", "VI → EN", "JA → EN", "EN → JA", "VI → JA", "JA → VI", "ZH → EN"],
  design:      ["Figma", "CSS", "Tailwind CSS", "Framer", "Adobe XD", "Sketch"],
  writing:     ["Technical Docs", "Blog", "API Docs", "User Guide", "SEO", "Copywriting"],
  data:        ["Python", "SQL", "Pandas", "PySpark", "dbt", "Tableau", "Power BI"],
  custom:      [],
};

/** Bảng màu badge theo thứ tự hiển thị của category — dùng vòng lặp nên áp dụng được cho danh mục mới thêm qua Master Data. */
export const CATEGORY_BADGE_PALETTE: string[] = [
  "bg-slate-100 text-slate-700",
  "bg-blue-100 text-blue-700",
  "bg-violet-100 text-violet-700",
  "bg-cyan-100 text-cyan-700",
  "bg-orange-100 text-orange-700",
  "bg-emerald-100 text-emerald-700",
  "bg-pink-100 text-pink-700",
  "bg-amber-100 text-amber-700",
  "bg-teal-100 text-teal-700",
];

File: src-tauri/src/models/skill.rs
//! Model cho thư viện Skill — chỉ dẫn tái sử dụng để "nạp" cho agent khi mở
//! một terminal session (dùng độc lập hoặc gắn vào 1 step của Workflow).

use serde::{Deserialize, Serialize};

fn default_skill_icon() -> String {
    "pi pi-book".to_string()
}

fn default_string() -> String {
    String::new()
}

/// Một skill — thư viện global, tái sử dụng cho mọi workspace/workflow.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct Skill {
    pub id: i64,
    pub name: String,
    #[serde(default)]
    pub description: String,
    #[serde(default = "default_skill_icon")]
    pub icon: String,
    /// Danh mục skill — tên (`name`) một mục `master_data` với `keygroup = "SKILL_TYPE"`.
    #[serde(default = "default_string")]
    pub category: String,
    /// Tech stack cụ thể, VD: "Vue 3", "FastAPI", "Flutter" ...
    #[serde(default = "default_string")]
    pub stack: String,
    /// Nội dung chỉ dẫn (markdown) — tương đương SKILL.md.
    #[serde(default)]
    pub instructions: String,
    #[serde(default)]
    pub tags: Vec<String>,
    pub created_at: String,
    pub updated_at: String,
}

/// Request tạo/cập nhật skill.
#[derive(Debug, Deserialize)]
pub struct SkillRequest {
    pub name: String,
    #[serde(default)]
    pub description: String,
    #[serde(default = "default_skill_icon")]
    pub icon: String,
    #[serde(default = "default_string")]
    pub category: String,
    #[serde(default = "default_string")]
    pub stack: String,
    #[serde(default)]
    pub instructions: String,
    #[serde(default)]
    pub tags: Vec<String>,
}
